### PR TITLE
Avoid calling average_parameters multiple times per optimizer step

### PR DESCRIPTION
**Pull request by wayi1**
## What does this PR do?

Fix model averaging, which is recently updated in https://github.com/PyTorchLightning/pytorch-lightning/pull/12378

See the bug that I reported in PyTorch repo: https://github.com/pytorch/pytorch/pull/74737

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/strategies/ddp.py [hunk 1]
```diff
@@ -280,11 +280,12 @@ def optimizer_step(
         if not _TORCH_GREATER_EQUAL_1_10 or self._model_averager is None:
             return optimizer_output
 
+        params = []
```
Comment (carmocca) [inline pytorch_lightning/strategies/ddp.py:283]: Please add an entry to CHANGELOG's "Fixed" section

**Review by justusschock — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -251,6 +251,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed `ModelCheckpoint.on_save_checkpoint` and `ModelCheckpoint.on_load_checkpoint` in favor of `ModelCheckpoint.state_dict` and `ModelCheckpoint.load_state_dict` ([#11887](https://github.com/PyTorchLightning/pytorch-lightning/pull/11887))
 - Removed `Timer.on_save_checkpoint` and `Timer.on_load_checkpoint` in favor of `Timer.state_dict` and `Timer.load_state_dict` ([#11887](https://github.com/PyTorchLightning/pytorch-lightning/pull/11887))
 - Replaced PostLocalSGDOptimizer with a dedicated model averaging component ([#12378](https://github.com/PyTorchLightning/pytorch-lightning/pull/12378))
+- Avoid calling `average_parameters` multiple times per optimizer step ([#12452](https://github.com/PyTorchLightning/pytorch-lightning/pull/12452))
```
Comment (justusschock) [inline CHANGELOG.md:254]: Can you move this to the 1.7 section as 1.6 is already out?

**Review by justusschock — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Borda) [reply CHANGELOG.md]: done

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Changed
```
Comment (carmocca) [inline CHANGELOG.md:26]: This should go in "Fixed", not "Changed"

(I cannot do the push myself)
